models.py

```diff
@@ -584,8 +607,27 @@ def forward(self, signal: shaping.BT, mask: shaping.BT = None, **kwargs) -> shap
 
 		power_spectrum = real_squared + imag_squared
 		log_mel_features = self.mel(power_spectrum).log()
+
+		'''
+		if self.satisfy_features_divisibility_by_32 and log_mel_features.shape[-1] % 32 != 0:
+			pad = int(math.ceil(log_mel_features.shape[-1] / 32) * 32) - log_mel_features.shape[-1]
+			log_mel_features = F.pad(log_mel_features, (0, pad), mode='constant', value = 0)
+		'''
+		if self.check_features_divisibility_by_32:
+			if signal.shape[-1] % (32 / 2) != 0:
+				warnings.warn('Shape of input signal not divisible by 16')
+
+			if log_mel_features.shape[-1] % 32 != 0:
+				warnings.warn('Shape of frontend output signal not divisible by 32')
+
 		return log_mel_features
 
+	@staticmethod
+	# used formula from https://pytorch.org/docs/stable/generated/torch.nn.Conv1d.html
+	def get_out_shape(l_in, kernel_size, stride, additional_padding, padding=0, dilation=1):
```

Comment:
get_out_shape -> compute_output_shape

`l_in` -> something more semantic. `l_in` is uncomprehensible by itself

why do you call it `additional_padding`? we're dealing with regular padding, right?

Reply:
done